src/surface_conditions/surface_conditions.jl

```diff
@@ -26,9 +26,7 @@ function update_surface_conditions!(Y, p, t)
         Fields.field_values(Fields.level(Fields.coordinate_field(Y.c).z, 1))
     sfc_conditions_values = Fields.field_values(sfc_conditions)
     wrapped_sfc_setup = sfc_setup_wrapper(sfc_setup)
-    sfc_temp_var =
-        p.atmos.surface_model isa PrognosticSurfaceTemperature ?
-        Fields.field_values(Y.sfc.T) : nothing
+    sfc_temp_var = get_sfc_temp_var(atmos, t, params, p, Y)
```

Comment:
This is not a good way to set the surface temperature: it is specialized code for this particular mode that makes everything more complicated.